plugins/outputs/sql/sql.go

```diff
@@ -200,28 +213,130 @@ func (p *SQL) generateInsert(tablename string, columns []string) string {
 		strings.Join(placeholders, ","))
 }
 
-func (p *SQL) tableExists(tableName string) bool {
-	stmt := strings.ReplaceAll(p.TableExistsTemplate, "{TABLE}", quoteIdent(tableName))
+func (p *SQL) fetchColumns(tableName string) error {
+	stmt := p.ColumnQueryTemplate
+	stmt = strings.ReplaceAll(stmt, "{TABLE}", quoteStr(tableName))
+
+	columns, err := p.db.Query(stmt)
+
+	if err != nil {
+		return err
+	}
+
+	defer columns.Close()
```

Comment:
Please do not randomly insert empty lines! They should be used sparsely as elements to separate logical operation blocks.
```suggestion
	columns, err := p.db.Query(stmt)
	if err != nil {
		return err
	}
	defer columns.Close()
```
Same for other instances below.